src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DataGridViewAddColumnDialog.cs

```diff
@@ -0,0 +1,1012 @@
+                {
+                    if (typeof(IList<object>).IsAssignableFrom(props[i].PropertyType))
+                    {
+                        // we have an IList. It could be a byte[] in which case we want to generate an Image column
+                        //
+                        TypeConverter imageTypeConverter = TypeDescriptor.GetConverter(typeof(Image));
+                        if (!imageTypeConverter.CanConvertFrom(props[i].PropertyType))
+                        {
+                            continue;
+                        }
+                    }
+
+                    _dataColumns.Items.Add(new ListBoxItem(props[i].PropertyType, props[i].Name));
+                }
+            }
+        }
+
+        if (selectedIndex != -1 && selectedIndex < _dataColumns.Items.Count)
+        {
+            _dataColumns.SelectedIndex = selectedIndex;
+        }
+        else
+        {
+            _dataColumns.SelectedIndex = _dataColumns.Items.Count > 0 ? 0 : -1;
+        }
+    }
+
+    private void addButton_Click(object? sender, EventArgs e)
+    {
+        _cancelButton!.Text = (SR.DataGridView_Close);
+
+        // AddColumn takes all the information from this dialog, makes a DataGridViewColumn out of it
+        // and inserts it at the right index in the data grid view column collection
+        AddColumn();
+    }
+
+    private void cancelButton_Click(object? sender, EventArgs e)
+    {
+        Close();
+        /*
+        if (this.cancelButtonIsCloseButton)
+        {
+            this.AddColumn();
+        }
+        */
+    }
+
+    protected override bool ProcessDialogKey(Keys keyData)
+    {
+        if ((keyData & Keys.Modifiers) == 0)
+        {
+            switch (keyData & Keys.KeyCode)
+            {
+                case Keys.Enter:
+                    // Validate the name before adding the column.
+                    IDesignerHost? host = null;
+                    INameCreationService? nameCreationService = null;
+                    IContainer? container = null;
+
+                    host = _liveDataGridView?.Site?.GetService(iDesignerHostType) as IDesignerHost;
+                    if (host is not null)
+                    {
+                        container = host.Container;
+                    }
+
+                    nameCreationService = _liveDataGridView?.Site?.GetService(iNameCreationServiceType) as INameCreationService;
+
+                    string errorString = string.Empty;
+                    if (ValidName(_nameTextBox!.Text,
+                                  _dataGridViewColumns!,
+                                  container,
+                                  nameCreationService,
+                                  _liveDataGridView!.Columns,
+                                  !_persistChangesToDesigner,
+                                  out errorString))
+                    {
+                        this.AddColumn();
+                        this.Close();
```

Comment:
Please remove the `this` modifier though out this PR. This could be done in  a separate commit or in a follow up PR
```suggestion
                        AddColumn();
                        Close();
```